tools/gdb/thread.py

```diff
@@ -0,0 +1,286 @@
+############################################################################
+# tools/gdb/thread.py
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.  The
+# ASF licenses this file to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+import gdb
+import utils
+
+UINT16_MAX = 0xFFFF
+
+g_tcbinfo = gdb.parse_and_eval("g_tcbinfo")
```

Comment:
let's remove global variables to avoid the cache value stale after user reload image.

Reply:
ok